app/utils/utils.ts

```diff
@@ -34,6 +35,50 @@ export function validateNewRuleset(newRuleset: Ruleset, rulesets: Ruleset[]): {
 
   return { isValid: true };
 }
+export function validateHours(newHoursWorked: EnterHours, hoursWorked: EnterHours[]): { isValid: boolean, errorMessage?: string } {
+
+  if (newHoursWorked.startTime == newHoursWorked.endTime) {
+    return { isValid: false, errorMessage: 'Start time and end time cannot be the same.' };
+  }
+
+  if (newHoursWorked.startTime > newHoursWorked.endTime) {
+    return { isValid: false, errorMessage: 'End time cannot be before start time.' };
+  }
+
+  const overlappingHours = hoursWorked.some((existingHours) => {
+    const newHoursStart = newHoursWorked.startTime;
+    const newHoursEnd = newHoursWorked.endTime;
+    const existingHoursStart = existingHours.startTime;
+    const existingHoursEnd = existingHours.endTime;
+
+    return (
+      newHoursWorked.date === existingHours.date &&
+      newHoursStart < existingHoursEnd &&
+      newHoursEnd > existingHoursEnd
+    );
+  });
+
+  if (overlappingHours) {
+    return { isValid: false, errorMessage: 'The new TOIL that you entered overlaps with hours that you have already added.' };
+  }
+
+  const exactMatch = hoursWorked.some((existingHours) => {
```

Comment:
I tried testing this scenario and I was able to submit the same data twice so there may be some issue in this implementation